source/client/process_impl.cc

```diff
@@ -433,5 +436,53 @@ void ProcessImpl::setupForHRTimers() {
   putenv(const_cast<char*>("EVENT_PRECISE_TIMER=1"));
 }
 
+RemoteProcessImpl::RemoteProcessImpl(const Options& options) : options_(options) {}
+
+bool RemoteProcessImpl::run(OutputCollector& collector) {
+  UriPtr uri;
+
+  try {
+    uri = std::make_unique<UriImpl>(options_.nighthawkService());
+  } catch (const UriException&) {
+    ENVOY_LOG(error, "Bad service uri: {}", options_.nighthawkService());
+    return false;
+  }
+
+  auto channel = grpc::CreateChannel(fmt::format("{}:{}", uri->hostWithoutPort(), uri->port()),
```

Comment:
I am wondering if it would be better to adjust the API of RemoteProcessImpl so that it takes the stub in by pointer (raw or otherwise). That way it will be agnostic to how the gRPC stub is created, with what credentials, etc. WDYT?

Reply:
That would be better, but it I presume it would also force handling of options outside of this class; to propose an alternative, maybe a `GrpcChannelFactory` concept would be useful here we'd pass that into the API, and we can use that to create the right channel for us?
Do you know if there a existing proto definition around somewhere which generalizes grpc channel configuration? (If so, maybe we can re-use that).